channels_redis/core.py

```diff
@@ -207,15 +220,23 @@ async def send(self, channel, message):
 
         # Check the length of the list before send
         # This can allow the list to leak slightly over capacity, but that's fine.
-        if await connection.zcount(channel_key, "-inf", "+inf") >= self.get_capacity(
-            channel
-        ):
+        current_length = await connection.zcount(channel_key, "-inf", "+inf")
+
+        if current_length + len(messages) > self.get_capacity(channel):
             raise ChannelFull()
 
         # Push onto the list then set it to expire in case it's not consumed
-        await connection.zadd(channel_key, {self.serialize(message): time.time()})
+        await connection.zadd(channel_key, mapping)
         await connection.expire(channel_key, int(self.expiry))
 
+    def _parse_messages(self, message):
+        """
+        Convert a passed message arg to a tuple of messages.
+        """
+        if not isinstance(message, dict) and hasattr(message, "__iter__"):
```

Comment:
Should we use `isinstance(message, collections.abc.Iterable` here instead? I think that would make this a bit more readable

Reply:
I can iterate a dict no?

Reply:
Indeed, python `dict` is an instance of `collections.abc.Iterable`